packages/selection/frontend/sirius-components-selection/src/SelectionDialog.tsx

```diff
@@ -174,7 +176,7 @@ export const SelectionDialog = ({
             color="primary"
             onClick={() => {
               if (selectedObjectId) {
-                onFinish(selectedObjectId);
+                onFinish([{ name: 'selectedObject', value: selectedObjectId, type: GQLToolVariableType.objectId }]);
```

Comment:
It seems you create a new `GQLToolVariable ` with the `selectedObjectId` as value, but you name it `selectedObject`. Why you don't name it `selectedObjectId`?

Reply:
The "specifier" will manipulate the object (the variable value will be the object and not its id)

Reply:
So the "operation" that will transform an _objectId_ to an _object_ is done on the backend right?
So why you have to create a variable here, in the frontend?

Reply:
The idea is that the backend part does not make any assumptions on the variables provided by the frontend component. For now, we have identified 3 kinds of values which can be provided by a Dialog: An Object: in that case we use the Id. An array of objects or a simple string.

Reply:
I get that, but it still seems strange.
Here you set the variable _type_ to `GQLToolVariableType.objectId` which indicates that the variable will contain an _objectId_, but you set the variable _name_  to `selectedObject` which indicates that the variable will contain an _Object_

Reply:
The type represents a technical information to describe which kind of value is set.
We can change the variable type by object instead of objectId, in that case, it will represent the value type once converted by the backend side.

Reply:
> Here you set the variable type to GQLToolVariableType.objectId which indicates that the variable will contain an objectId, but you set the variable name to selectedObject which indicates that the variable will contain an Object

No that's not how we should think about this. As the contributor of a dialog, I want my dialog to be used to create a variable named `selectedObject`. To populate this variable, I'll give the backend some string as the value and it should be interpreted as the identifier of an object to resolve.